src/test/rustdoc-ui/reference-link-reports-error-once.rs

```diff
@@ -0,0 +1,8 @@
+#![deny(broken_intra_doc_links)]
+
+/// Links to [a] [link][a]
+/// And also a [third link][a]
```

Comment:
Hmm, I think I remember you having more test cases in https://github.com/rust-lang/rust/pull/77700. Can't remember what they were, though.

Reply:
This kinda belongs on #77700, but I removed the tests when we narrowed the PR's scope, because adding a cache shouldn't change behaviour and they didn't belong there any more.